src/app/dataValidator/index.js

```diff
@@ -9,8 +10,43 @@ const validateData = data => {
   log('\nValidation complete!');
 };
 
-const data = require('../../../data/scenario-01.json');
+const validateFile = filename => {
+  const data = fs.readFileSync(filename);
+  validateData(JSON.parse(data));
+};
+
+const readAllFiles = (filenames, dirname) =>
+  filenames.forEach(filename => {
+    // explicitly ignore scenario-23
+    if (filename.includes('scenario-23.json')) {
+      return;
+    }
+
+    if (filename.includes('.json')) {
+      validateFile(`${dirname}/${filename}`);
+    }
+  });
 
-validateData(data);
+const readdirAsync = dirname =>
+  new Promise((resolve, rejects) => {
+    fs.readdir(dirname, (err, filenames) => {
+      if (err) {
+        rejects(err);
+      } else {
+        resolve(readAllFiles(filenames, dirname));
+      }
+    });
+  });
 
-module.exports.validateData = validateData;
+console.time('readdirAsync'); // eslint-disable-line no-console
+readdirAsync('./././data').then(() => {
+  console.timeEnd('readdirAsync'); // eslint-disable-line no-console
+  log('\nAll files validated!');
+});
+
+module.exports = {
```

Comment:
I think in this file you should only export a single function. 
If you want to export each of these for testing purposes, then they should be in separate files.

Reply:
Yes, we could use something like [rewire](https://github.com/jhnns/rewire) if we need to test internals. Unsure if there is anything newer in the new Jest world (Rewire is testing lib agnostic though).

Reply:
The commit https://github.com/bbc/simorgh/pull/340/commits/1125839d944889df7279fa461f2063bb7eb7c44a splits these out into more granular methods and which makes the testing/exporting more explicit. 
One part to mention is where I prepend `module.exports.` to the method call (https://github.com/bbc/simorgh/pull/340/commits/1125839d944889df7279fa461f2063bb7eb7c44a#diff-82c901e3afd825f4a1151b3b0f510687R13) this is needed otherwise the spy can't track calls to a method in the current module.